lldb/source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -2201,25 +2201,26 @@ SwiftLanguageRuntime::GetRuntimeUnwindPlan(ProcessSP process_sp,
   Address pc;
   pc.SetLoadAddress(regctx->GetPC(), &target);
   SymbolContext sc;
-  if (pc.IsValid()) {
-    pc.CalculateSymbolContext(&sc,
-                              eSymbolContextFunction | eSymbolContextSymbol);
-    if (sc.function) {
-      Address func_start_addr = sc.function->GetAddressRange().GetBaseAddress();
-      AddressRange prologue_range(func_start_addr,
-                                  sc.function->GetPrologueByteSize());
-      if (prologue_range.ContainsLoadAddress(pc, &target) ||
-          func_start_addr == pc) {
-        return UnwindPlanSP();
-      }
-    } else if (sc.symbol) {
-      Address func_start_addr = sc.symbol->GetAddress();
-      AddressRange prologue_range(func_start_addr,
-                                  sc.symbol->GetPrologueByteSize());
-      if (prologue_range.ContainsLoadAddress(pc, &target) ||
-          func_start_addr == pc) {
-        return UnwindPlanSP();
-      }
+  if (pc.IsValid())
+    if (!pc.CalculateSymbolContext(&sc, eSymbolContextFunction |
+                                            eSymbolContextSymbol))
+      return UnwindPlanSP();
+
+  if (sc.function) {
+    Address func_start_addr = sc.function->GetAddressRange().GetBaseAddress();
+    AddressRange prologue_range(func_start_addr,
+                                sc.function->GetPrologueByteSize());
+    if (prologue_range.ContainsLoadAddress(pc, &target) ||
+        func_start_addr == pc) {
+      return UnwindPlanSP();
+    }
+  } else if (sc.symbol) {
+    Address func_start_addr = sc.symbol->GetAddress();
+    AddressRange prologue_range(func_start_addr,
+                                sc.symbol->GetPrologueByteSize());
+    if (prologue_range.ContainsLoadAddress(pc, &target) ||
+        func_start_addr == pc) {
+      return UnwindPlanSP();
```

Comment:
is this if-else cascade exhaustive or should we add a catch-all else at the end to also return an empty unwind plan and perhaps log an error?

Reply:
I originally considered an `else` here, but went with this change because it's catching the issue at the source (the return value of `CalculateSymbolContext`. Maybe an `else` would be more clear? But I think we only need one or the other, not both.

Reply:
Is the different line the one with `func_start_addr =`? That could be a parameter for the lambda. Let me know if I'm overlooking something else.

Reply:
yep that's the one line, you're not overlooking anything.